pulumi/src/hash.ts

```diff
@@ -0,0 +1,133 @@
+import { createHash } from 'crypto'
+import { parse } from 'dotenv'
+import { hashElement } from 'folder-hash'
+import objectHash from 'object-hash'
+import * as k8s from '@pulumi/kubernetes'
+
+export enum CoinstackType {
+  GO,
+  NODE,
+}
+
+export const secretEnvs = (assetName: string, sampleEnv: Buffer) =>
+  Object.keys(parse(sampleEnv)).map<k8s.types.input.core.v1.EnvVar>((key) => ({
+    name: key,
+    valueFrom: { secretKeyRef: { name: assetName, key: key } },
+  }))
+
+export const getCoinstackHash = async (
+  coinstack: string,
+  buildArgs: Record<string, string>,
+  coinstackType: CoinstackType
+): Promise<string> => {
+  switch (coinstackType) {
+    case CoinstackType.NODE:
+      return await getNodeCoinstackHash(coinstack, buildArgs)
+    case CoinstackType.GO:
+      return await getGoCoinstackHash(coinstack, buildArgs)
+  }
+}
+
+// creates a hash of the content included in the final build image
+const getNodeCoinstackHash = async (coinstack: string, buildArgs: Record<string, string>): Promise<string> => {
+  const hash = createHash('sha1')
+  const nodeBasePath = `${__dirname}/../../node`
+
+  // hash root level unchained files
+  const { hash: unchainedHash } = await hashElement(nodeBasePath, {
```

Comment:
this needs to be the root dir and also include the `yarn.lock` and `Dockerfile.node`